bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.server;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ism8Server} is responsible for listening to the Ism8 information
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Server extends Thread {
+    private final Logger logger = LoggerFactory.getLogger(Server.class);
+
+    private int port;
+    private int startRetries;
+    private boolean connected;
+    private ArrayList<IDataPoint> dataPoints = new ArrayList<>();
+    private @Nullable ServerSocket serverSocket = null;
+    private @Nullable Socket client;
+    private @Nullable IDataPointChangeListener changeListener;
+
+    public Server(int port) {
+        this.port = port;
```

Comment:
Please use named threads and set it to daemon.
```suggestion
        super("OH-binding-<insert the ThingUID>");
        setDaemon(true);
        this.port = port;
```

Reply:
I was trying to find any reference for passing the ThingUID, but couldn't find any good reference. Only one existing binding is using somthing similar, so I took this as reference. Basically I'm inserting the ThingUID as parameter in the constructor in the handler after getting this by using:
`final String uid = this.getThing().getUID().getAsString();` 
`Server svr = new Server(cfg.getPortNumber(), uid);`
`public Server(int port, String uid)` 
` super("OH-binding-" + uid);.`
Is this the correct way?

Reply:
That looks good.